src/subscribe/tests.rs

```diff
@@ -262,5 +263,5 @@ async fn timeseries_with_conn() {
         min += THREE_MIN;
     }
 
-    assert_eq!(timeseries.series[36], 3.0);
+    assert!((timeseries.series[36] - 3.0).abs() < ALLOWED_ERROR_EVENT_COUNT);
```

Comment:
When comparing floating-point numbers, we prefer using the `partial_cmp` method.

Reply:
Defining an arbitrary margin for comparison at the user's discretion is clearly inappropriate. Whenever we feel that our approach doesn't perfectly suit the current problem, it's time to consult Rust's standard library.

Reply:
Thank you for the feedback. The original Clippy warning is likely about the incorrect representation of floating-point numbers in computer systems. For example, the following code produces an erroneous result:

```rust
fn main() {
    let a: f64 = 1000.1;
    let b: f64 = 0.2;
    let y = 1000.3; // Expected value.

    let are_equal = (a + b).partial_cmp(&y).unwrap();
    println!("{are_equal:?}"); // Greater
}
```

I'm curious whether the convention we use takes data inaccuracies into account. In this case, using `partial_cmp` to suppress the Clippy warning seems practical, as the data is effectively treated as integers.

Reply:
I think we can take benefit of std lib's `f64::EPSILON` as the precision tolerance. So something like

```rust
(a - b).abs() < f64::EPSILON
```

Reply:
The [Clippy documentation](https://rust-lang.github.io/rust-clippy/master/index.html#float_cmp) about the warning mentions not to use `EPSILON`, as shown below 😅
<img width="1204" alt="image" src="https://github.com/user-attachments/assets/2559faa9-2f69-45b2-8883-609801efc257" />

Reply:
Thank you for the link. It helped me understand the definition of EPSILON better. And upon reading it, I came to think that we can adopt the Clippy's recommendation (https://floating-point-gui.de/errors/comparison/ ) like below:

```rust
// if we use `<` 
(a - b).abs().div(f64::MAX) < f64::EPSILON

// if we use `partial_cmp`
(a - b).abs().div(f64::MAX).partial_cmp(&f64::EPSILON) == Some(std::cmp::Ordering::Less)
```

The main reason behind this is that we can use the logic in the last `else` block below, and ignore the `if` and `else if` blocks, because I believe those are not the cases we are facing now.

```cpp
// This is the code from https://floating-point-gui.de/errors/comparison/
public static boolean nearlyEqual(float a, float b, float epsilon) {
	final float absA = Math.abs(a);
	final float absB = Math.abs(b);
	final float diff = Math.abs(a - b);

	if (a == b) { // shortcut, handles infinities
		return true;
         } else if (a == 0 || b == 0 || (absA + absB < Float.MIN_NORMAL)) {
		// a or b is zero or both are extremely close to it
		// relative error is less meaningful here
		return diff < (epsilon * Float.MIN_NORMAL);
	} else { // use relative error
		return diff / Math.min((absA + absB), Float.MAX_VALUE) < epsilon;
	}
}
```

Reply:
I don't think we need to put in such precise effort to write test code. Besides, this comparison is only for test and is not part of the main logic. Therefore, simplicity is best here.

Reply:
Applied the suggested adjustment.